modulus/distributed/manager.py

```diff
@@ -287,6 +287,39 @@ def initialize_open_mpi(addr, port):
             method="openmpi",
         )
 
+    @staticmethod
+    def initialize_mpich(addr, port):
+        """Setup method using MPICH initialization"""
+        rank = int(os.environ.get("PMI_RANK"))
+        world_size = int(os.environ.get("PMI_SIZE"))
+
+        # cray-mpich
+        if "PMI_LOCAL_RANK" in os.environ:
+            local_rank = int(os.environ.get("PMI_LOCAL_RANK"))
+        # mpich-4.2.1 / hydra
+        else:
+            local_rank = int(os.environ.get("MPI_LOCALRANKID"))
+
+        # for multi-node MPI jobs, determine "addr" as the
+        # address of global rank 0.
+        if "localhost" == addr:
```

Comment:
Nit: we don't really use Yoda conditions in Modulus code since assignment to var is a syntax error (unless you use walrus op which has a different syntax).